### ISSUE TITLE
Don't load 3rd party JS by default

### ISSUE THREAD

**Issue by yuvipanda** [notebook-sharing-space/nbss#7]
We load 3rd party JS for a *lot* of things. We can probably replace all of them, except hypothesis. We should. This might mean we need webpack tho.

**Comment by yuvipanda** [comment]
Mathjax and requirejs are the only ones left now. However, rendered notebooks will probably still load 3rd party JS. Nothing much I can do about that.

**Comment by psychemedia** [comment]
Re: rendered notebooks - I guess it would be possible to sanitise and maybe even rewrite to a whitelisted set domains, although that would mean rewriting uploaded ipynb?

**Comment by yuvipanda** [comment]
Ah, hmm. I was only thinking of the things we load by default - requirejs and mathjax. 

This brings up a question of how safe we want to consider the rendered notebook to be. GitHub runs no JS, nbviewer does. Currently, we're at the nbviewer level - running hypothes.is is already unsafe now....

**Comment by psychemedia** [comment]
To my mind, nbviewer rendering is far more useful than Github (plus Github renderer continues to have issues accessing notebook files...).

If you are looking at JupyterBook levels of rendering, then folk absolutely need ability to embed js powered outputs?

**Comment by yuvipanda** [comment]
yeah totally! I think I'll do this in a way where if we enable hypothesis, inetractivity in notebook stops. Otherwise it can keep going. This is a good compromise I think

**Comment by psychemedia** [comment]
In passing, I note this (old?) site relating to Hypothes.is security model: https://h.readthedocs.io/projects/client/en/latest/developers/security/
